### PR TITLE
Parser for DPLL samples

### PR DESCRIPTION
**Pull request by spenced**

### CODE REVIEW

File: src/vse_sync_pp/parse.py [hunk 1]
```diff
@@ -9,10 +9,12 @@
 import json
 from decimal import Decimal
 
+from .parsers import dpll
 from .parsers import ts2phc
 
 PARSERS = {
     cls.id_: cls for cls in (
+        dpll.TimeOffset,
```
Comment (jnunyez) [inline src/vse_sync_pp/parse.py:17]:
```suggestion
        dpll.PhaseOffset,
```

File: tests/vse_sync_pp/parsers/test_dpll.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Test cases for vse_sync_pp.parsers.dpll"""
+
+from unittest import TestCase
+from decimal import Decimal
+
+from vse_sync_pp.parsers.dpll import (
+    TimeOffset,
```
Comment (jnunyez) [inline tests/vse_sync_pp/parsers/test_dpll.py:9]:
```suggestion
    PhaseOffset,
```

File: tests/vse_sync_pp/parsers/test_dpll.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+from .test_parser import ParserTestBuilder
+
+class TestTimeOffset(TestCase, metaclass=ParserTestBuilder):
+    """Test cases for vse_sync_pp.parsers.dpll.TimeOffset"""
+    constructor = TimeOffset
+    id_ = 'dpll/time-offset'
+    elems = ('timestamp', 'eecstate', 'phasestate', 'toffset')
```
Comment (jnunyez) [inline tests/vse_sync_pp/parsers/test_dpll.py:18]:
```suggestion
    elems = ('timestamp', 'eecstate', 'phasestate', 'phaseoffset')
```

File: src/vse_sync_pp/parsers/dpll.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Parse dpll samples"""
+
+from collections import namedtuple
+from decimal import (Decimal, InvalidOperation)
+
+class TimeOffset():
+    """Parse time offset from a dpll sample"""
+    id_ = 'dpll/time-offset'
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/dpll.py:10]:
```suggestion
    id_ = 'dpll/phase-offset'
```

File: src/vse_sync_pp/parsers/dpll.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    elems = ('timestamp', 'eecstate', 'phasestate', 'toffset')
+    parsed = namedtuple('Parsed', elems)
+    def parse(self, line):
+        """Parse time offset from `line`.
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/dpll.py:14]:
```suggestion
        """Parse phase offset from `line`.
```

Comment (jnunyez) [inline src/vse_sync_pp/parsers/dpll.py:14]:
```suggestion
        """Parse phase offset from `line`.
```

File: src/vse_sync_pp/parsers/dpll.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            timestamp = Decimal(elems[0])
+        except InvalidOperation as exc:
+            raise ValueError(elems[0]) from exc
+        try:
+            toffset = Decimal(elems[3])
+        except InvalidOperation as exc:
+            raise ValueError(elems[3]) from exc
+        eecstate = int(elems[1])
+        phasestate = int(elems[2])
+        return self.parsed(timestamp, eecstate, phasestate, toffset)
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/dpll.py:33]:
```suggestion
        return self.parsed(timestamp, eecstate, phasestate, phaseoffset)
```

**Review by jnunyez — COMMENTED** [review]
Added a suggestion  about the DPLL phase offset that we are calculating (i.e., phase offset instead of time offset).

File: src/vse_sync_pp/parsers/dpll.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+"""Parse dpll samples"""
+
+from collections import namedtuple
+from decimal import (Decimal, InvalidOperation)
+
+class TimeOffset():
+    """Parse time offset from a dpll sample"""
+    id_ = 'dpll/time-offset'
+    elems = ('timestamp', 'eecstate', 'phasestate', 'toffset')
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/dpll.py:11]:
```suggestion
    elems = ('timestamp', 'eecstate', 'phasestate', 'phaseoffset')
```

File: tests/vse_sync_pp/parsers/test_dpll.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+from unittest import TestCase
+from decimal import Decimal
+
+from vse_sync_pp.parsers.dpll import (
+    TimeOffset,
+)
+
+from .test_parser import ParserTestBuilder
+
+class TestTimeOffset(TestCase, metaclass=ParserTestBuilder):
```
Comment (jnunyez) [inline tests/vse_sync_pp/parsers/test_dpll.py:14]:
```suggestion
class TestPhaseOffset(TestCase, metaclass=ParserTestBuilder):
```

**Comment by spenced** [comment]
Restored unit tests to pass; squash.

**Review by crwr45 — APPROVED** [review]

**Pull request merged**